api/src/main/java/com/bookwhale/common/token/JWT.java

```diff
@@ -0,0 +1,259 @@
+        this.algorithm = Algorithm.HMAC512(clientSecret);
+        this.jwtVerifier = com.auth0.jwt.JWT.require(algorithm).withIssuer(issuer).build();
+    }
+
+    public String createNewToken(Claims claims) {
+        LocalDateTime now = LocalDateTime.now();
+        Instant instant = now.atZone(ZoneId.systemDefault()).toInstant();
+        Date current = Date.from(instant);
+
+        JWTCreator.Builder tokenBuilder = com.auth0.jwt.JWT.create();
+        tokenBuilder.withIssuer(issuer);
+        tokenBuilder.withIssuedAt(current);
+
+        if (expiryMilliSeconds > 0) {
+            tokenBuilder.withExpiresAt(new Date(instant.toEpochMilli() + expiryMilliSeconds));
+        }
+
+        tokenBuilder.withClaim("name", claims.name);
+        tokenBuilder.withClaim("email", claims.email);
+        tokenBuilder.withClaim("image", claims.image);
+
+        return tokenBuilder.sign(algorithm);
+    }
+
+    public String createNewRefreshToken(ClaimsForRefresh claims) {
+        LocalDateTime now = LocalDateTime.now();
+        Instant instant = now.atZone(ZoneId.systemDefault()).toInstant();
+        Date current = Date.from(instant);
+
+        JWTCreator.Builder tokenBuilder = com.auth0.jwt.JWT.create();
+        tokenBuilder.withIssuer(issuer);
+        tokenBuilder.withIssuedAt(current);
+
+        if (expiryRefreshMilliSeconds > 0) {
+            tokenBuilder.withExpiresAt(
+                new Date(instant.toEpochMilli() + expiryRefreshMilliSeconds));
+        }
+
+        tokenBuilder.withClaim("rid", claims.rid);
+        tokenBuilder.withClaim("email", claims.email);
+
+        return tokenBuilder.sign(algorithm);
+    }
+
+    public String refreshApiToken(String token) throws JWTVerificationException {
+        DecodedJWT decodedJWT = com.auth0.jwt.JWT.decode(token);
+        Claims claims = new Claims(decodedJWT);
+        claims.removeIssuedAt();
+        claims.removeExpiresAt();
+        return createNewToken(claims);
+    }
+
+    public Claims verify(String token) {
+        DecodedJWT verify = null;
+        try {
+            verify = jwtVerifier.verify(token);
+        } catch (AlgorithmMismatchException e) {
+            log.error("not defined algorithm used.", e);
+            throw new CustomException(ErrorCode.FORBIDDEN);
+        } catch (SignatureVerificationException e) {
+            log.error("invalid signature.", e);
+            throw new CustomException(ErrorCode.FORBIDDEN);
+        } catch (TokenExpiredException e) {
+            log.error("token has expired.", e);
+            throw new CustomException(ErrorCode.FORBIDDEN);
+        } catch (InvalidClaimException e) {
+            log.error("not expected claim contained.", e);
+            throw new CustomException(ErrorCode.FORBIDDEN);
+        } catch (JWTVerificationException e) {
+            log.error("token verify failed.", e);
+            throw new CustomException(ErrorCode.FORBIDDEN);
+        }
```

Comment:
>AlgorithmMismatchException – if the algorithm stated in the token's header it's not equal to the one defined in the JWTVerifier.
SignatureVerificationException – if the signature is invalid.
TokenExpiredException – if the token has expired.
InvalidClaimException – if a claim contained a different value than the expected one